src/verbs/rdma_context.rs

```diff
@@ -0,0 +1,153 @@
+use crate::verbs::address_handle::Gid;
+use std::io;
+use std::mem::MaybeUninit;
+use std::ptr::{self, null_mut};
+
+use rdma_mummy_sys::{
+    ___ibv_query_port, ibv_alloc_pd, ibv_close_device, ibv_context, ibv_create_cq, ibv_create_cq_ex,
+    ibv_device_attr_ex, ibv_port_attr, ibv_query_device_ex, ibv_query_gid, ibv_query_gid_type,
+};
+
+use super::completion_queue::{CompletionQueue, CompletionQueueEx, CqInitAttr};
+use super::protection_domain::ProtectionDomain;
+
+pub struct RdmaContext {
```

Comment:
I just realized that `DeviceContext` may be a better name for this struct. What do you think?

Reply:
OK, I would rename this to `DeviceContext`